lib/sqlalchemy/orm/events.py

```diff
@@ -2883,7 +3079,9 @@ def my_event(query):
 
         """
 
-    def before_compile_update(self, query, update_context):
+    def before_compile_update(
+        self, query: Query[Any], update_context: _O
```

Comment:
**mike bayer** (zzzeek) wrote:

update_context here would be sqlalchemy.orm.query.BulkUpdate

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4326